### PR TITLE
Add precommit automation

### PR DESCRIPTION
**Pull request by anishasthana**
- Add pre-commit configs for repository
- Add OWNERS file
- Add precommit workflow

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,10 +19,14 @@ Can be installed via `pip`: `pip install codeflare-sdk`
 
 We recommend using Python 3.8 for development.
 Install development specific dependencies:
-  `$ pip install poetry pytest pytest-mock coverage black==22.3.0`
+  `$ pip install pre-commit poetry pytest pytest-mock coverage black==22.3.0`
```
Comment (KPostOffice) [inline README.md:22]: Could we put these in a requirements-dev.txt file? and then just do `$ pip install -r requirements-dev.txt`?

**Review by KPostOffice — COMMENTED** [review]
This looks good to me, just one comment

File: README.md (hunk 1 above)
Comment (anishasthana) [reply README.md]: Done!

**Review by Maxusmusti — APPROVED** [review]
LGTM, a fan of these changes

**Review by Maxusmusti — CHANGES_REQUESTED** [review]
Seems pre-commit check is broken though

**Pull request merged**